docs/design/connection-tester.md

```diff
@@ -0,0 +1,50 @@
+# Connection Tester
+
+The goal of this document is to sketch out possible approaches for implementing _Connection testing_ in the context of iPaaS. This backend concept is introduced to support the use case to quickly test whether a given `Connection`  (which is the combination of a `Connector` and a `ConnectorConfiguration`) is actually usable. The typical use case is, that the user pushes a button "Test connection" on the "Connection Details" page which actually triggers the test of the connection and that can be either successful or failing. When it fails, a detailed error message describes not only the error itself but also how to resolve the error (e.g. "Cannot connect because of invalid credentials"). 
+
+## Communication flow
+
+When the user triggers this test function from within the UI, a backend call to `ipaas-rest` will be issued, which in turn will trigger a (still to be defined) method on the referenced connector with the concrete configuration given. This method, let's call it `ping()`, must be side effect free, which should be possible for most endpoints. The method takes the configuration as argument for the connector and returns a result objects which contains the result of the check (ok / nok) and in case of a failure the error message (question: Should we already think about I18N here ?)
```

Comment:
@KurtStam actually, that's a very good point about changing endpoints. But I didn't think its the citizen developers task to detect this, because what can he do if he finds out that the connector is broken ? Its the operator's task (== we) that has to monitor the endpoints about changes. Also, the change most likely affects existing integrations not when you create a new one. But agreed, its an important thing to monitor.

The "connector test" happens without any integration or data mapping, but when you define a `Connection` (== `Connector` + user specific `ConnectorConfiguration`). The integration comes after this. The API contract between ipaas-rest and the `Connector` should be the same for every connector, each connector should know on its own how to verify a configuration. So its independent from any connector specific payload (which we cannot provide anyway when we define a connection, where the "Test connection" button is supposed to be placed).